cns/requestcontroller/kubecontroller/crdtranslator.go

```diff
@@ -1,19 +1,74 @@
 package kubecontroller
 
 import (
+	"net"
+
 	"github.com/Azure/azure-container-networking/cns"
 	nnc "github.com/Azure/azure-container-networking/nodenetworkconfig/api/v1alpha"
 )
 
-// CRDStatusToNCRequest translates a crd status to network container request
-func CRDStatusToNCRequest(crdStatus *nnc.NodeNetworkConfigStatus) (*cns.CreateNetworkContainerRequest, error) {
-	//TODO: Translate CRD status into network container request
-	//Mat will pick up from here
-	return nil, nil
+// CRDStatusToNCRequest translates a crd status to createnetworkcontainer request
+func CRDStatusToNCRequest(crdStatus nnc.NodeNetworkConfigStatus) (cns.CreateNetworkContainerRequest, error) {
```

Comment:
Well, this returns a pointer. As ncRequest can be nil if no CRD exists. Sorry my earlier feedback mislead you. Remove pointers where it is not required.

Reply:
Can we write UT for these functions? Wondering how we will pass nnc.NodeNetworkConfigStatus

Reply:
I'll write UTs